apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ReliableStreamReader.cs

```diff
@@ -127,28 +128,42 @@ private async Task<bool> ReliableConnectAsync(CancellationToken cancellationToke
             if (_currentCall == null)
             {
                 await WithTiming(ConnectAsync(), "ConnectAsync").ConfigureAwait(false);
-                Debug.Assert(_currentCall != null, "_currentCall != null");
+                GaxPreconditions.CheckState(_currentCall != null, "Failed to connect");
 
-                for (int i = 0; i <= _resumeSkipCount; i++)
+                bool success = false;
+                try
                 {
-                    ThrowIfCancellationRequested(cancellationToken);
-
-                    _isReading = await WithTiming(
-                            _currentCall.ResponseStream.MoveNext(cancellationToken)
-                                .WithSessionExpiryChecking(_session),
-                            "ResponseStream.MoveNext")
-                        .ConfigureAwait(false);
-                    if (!_isReading || _currentCall.ResponseStream.Current == null)
+                    for (int i = 0; i <= _resumeSkipCount; i++)
```

Comment:
I'm not sure, but I think `_resumeSkipCount` is not needed, from what I get from the [docs](https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.v1#google.spanner.v1.PartialResultSet) there's a `resume_token` in all `PartialResultSet`.
And actually I think the new implementation wouldn't work if `_resumeSkipCount` were not always zero, because we would be skipping results that come after the result that the server returned for the `_resumeToken`, which we are asuming is the "correct one".
I might not be explaining myself very well here.

Reply:
I think it would be good to clarify this with the Spanner team in terms of expectations - but I *think* we obey resumeSkipCount correctly. I can add a test for that if it turns out that there isn't always a resume token.

Reply:
OK, that sounds good.